src/zkevm_specs/evm_circuit/execution/error_gas_uint_overflow.py

```diff
@@ -0,0 +1,167 @@
+            Opcode.LOG0,
+            Opcode.LOG1,
+            Opcode.LOG2,
+            Opcode.LOG3,
+            Opcode.LOG4,
+            Opcode.SHA3,
+            Opcode.EXP,
+            Opcode.MLOAD,
+            Opcode.MSTORE,
+            Opcode.MSTORE8,
+            Opcode.RETURN,
+            Opcode.REVERT,
+        ),
+    )
+    is_create = is_create_flag + is_create2_flag
+    is_dynamic_gas = (
+        is_calldatacopy
+        + is_codecopy
+        + is_extcodecopy
+        + is_returndatacopy
+        + is_sha3
+        + is_call
+        + is_delegatecall
+        + is_staticcall
+        + is_create_flag
+        + is_create2_flag
+        + is_log0
+        + is_log1
+        + is_log2
+        + is_log3
+        + is_log4
+        + is_mload
+        + is_mstore
+        + is_mstore8
+        + is_return
+        + is_revert
+    )
+    is_opcode_memory_size_overflow = (
+        is_safe_mul_overflow
+    ) = (
+        is_call_gas_cost_overflow
+    ) = (
+        is_non_zero_calldata_gas_overflow
+    ) = is_zero_calldata_gas_overflow = is_eip3860_overflow = FQ(0)
+
+    # IntrinsicGas
+    # https://github.com/ethereum/go-ethereum/blob/b946b7a13b749c99979e312c83dce34cac8dd7b1/core/state_transition.go#L67
+    calldata_length = instruction.call_context_lookup(CallContextFieldTag.CallDataLength)
+    tx_id = instruction.call_context_lookup(CallContextFieldTag.TxId)
+    is_root = instruction.call_context_lookup(CallContextFieldTag.IsRoot)
+
+    if is_root.expr() == FQ(1):
+        data = [
+            instruction.tx_calldata_lookup(tx_id, FQ(idx))
+            for idx in range(0, calldata_length.expr().n)
+        ]
+        data_len = len(data)
+
+        def transaction_data_gas_overflow() -> tuple[bool, bool]:
+            # zero and non-zero bytes are priced differently
+            nz = len([byte for byte in data if byte != 0])
+            gas = TxGasContractCreation if is_create == FQ(1) else TxGas
+            is_non_zero_calldata_gas_overflow, _ = instruction.compare(
+                FQ(((MAX_U64 - gas) // TxDataNonZeroGasEIP2028)), FQ(nz), N_BYTES_U64
+            )
+            gas += nz * TxDataNonZeroGasEIP2028
+
+            # tx data zero gas overflow
+            is_zero_calldata_gas_overflow = FQ(0)
+            if is_non_zero_calldata_gas_overflow == FQ(0):
+                z = data_len - nz
+                is_zero_calldata_gas_overflow, _ = instruction.compare(
+                    FQ(((MAX_U64 - gas) // TxDataZeroGas)), FQ(z), N_BYTES_U64
+                )
+
+            # TODO: Would like to support EIP 3860 in the future (See
+            # https://github.com/privacy-scaling-explorations/zkevm-specs/issues/421)
+            # gas += z * TxDataZeroGas
+            # if is_create:
+            #     lenWords = data_len // 32
+            #     is_eip3860_overflow, _ = instruction.compare(
+            #         FQ((MAX_U64 - gas) // InitCodeWordGas), FQ(lenWords), N_BYTES_U64
+            #     )
```

Comment:
Are these leftover or part of the TODO comments?

Reply:
It looks like a part of TODO comments. Do you think we need to remove it?

Reply:
I think we can remove them so that people won't get confused.